Source/Core/loadWithXhr.js

```diff
@@ -192,6 +192,8 @@ define([
             //or do not support the xhr.response property.
             if (defined(response) && (!defined(responseType) || (browserResponseType === responseType))) {
                 deferred.resolve(response);
+            } else if (xhr.status === 204) { // accept no content
```

Comment:
I missed this in the original review.  This check should happen first, not second.  This way is causing 204 to still resolve to `""`.  We need to also update the comment at the top which is now outdated.